### DISCUSSION TITLE
Does the CYD support HID Keyboards?

### DISCUSSION THREAD

**Discussion by Jackolix** [witnessmenow/ESP32-Cheap-Yellow-Display#215] [Q&A]
Can i use the CYD to send Keyboard commands to a connected PC? Ive tried the Adrduino HID library but it is not compatible with this board and ive read something that it has to be supported by the usb controller on the board(?). 

Any help is appreciated.
Thanks

**Comment by clach04** [comment]
I vaguely recall reading the newer ESP32 devices can handle HID with the build in USB port... i.e. not the one in the CYD?

If HID support is possible, I'd be very interested in this. However, you may be able to use the Bluetooth radio as a HID device instead.

**Comment by itarozzi** [comment]
I'm afraid it's not possible as it uses a CH340C to convert serial TTL to USB. So the USB port is not directly connected and not managed by ESP32 MCU.
